### PR TITLE
snapcraft.yaml: add minimal snapcraft.yaml with custom build

### PR DESCRIPTION
**Pull request by mvo5**
This adds a minimal snapcraft.yaml and a custom build plugin.
 The snapcraft.yaml only puts the bits into the snapd snap
    for now. This means we can start testing the core18/snapd
    interplay.
    
The plugin will just reuse the deb building (which we have to
    maintain anyway). The plugin should probably be cleaned up and
    pushed upstream as this seems to be useful for projects that
    already have a (complicated) deb based build setup.
    
 One issue with the current snapcraft is that it does not deal with
    a source dir called "snap" well. We have code located there but
    snapcraft will ignore this dir when pulling code or when setting
    up the build tree: https://bugs.launchpad.net/bugs/1772584

### CODE REVIEW

File: parts/plugins/x_builddeb.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# cowboy baby, see https://bugs.launchpad.net/snapcraft/+bug/1772584
+def patch_snapcraft():
+    import snapcraft.internal.common
+    # very hacky but gets the job done for now, right now
+    # SNAPCRAFT_FILES is only used to know what to exclude
+    snapcraft.internal.common.SNAPCRAFT_FILES.remove("snap")
+patch_snapcraft()
+
+
```
Comment (zyga) [inline parts/plugins/x_builddeb.py]: Nitpick, two newlines between functions.

File: snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+summary: Daemon and tooling that enable snap packages
+description: |
+ Install, configure, refresh and remove snap packages. Snaps are
+ 'universal' packages that work across many different Linux systems,
+ enabling secure distribution of the latest apps and utilities for
+ cloud, servers, desktops and the internet of things.
+
+ This is the CLI for snapd, a background service that takes care of
+ snaps on the system. Start with 'snap list' to see installed snaps.
+version: set-by-version-script-thxbye
```
Comment (zyga) [inline snapcraft.yaml]: It feels silly that we need to have this while having `version-script`

Comment (mvo5) [reply snapcraft.yaml]: Yeah, snapcraft is silly here.

File: snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: snapd
+summary: Daemon and tooling that enable snap packages
+description: |
+ Install, configure, refresh and remove snap packages. Snaps are
+ 'universal' packages that work across many different Linux systems,
+ enabling secure distribution of the latest apps and utilities for
+ cloud, servers, desktops and the internet of things.
+
+ This is the CLI for snapd, a background service that takes care of
```
Comment (stolowski) [inline snapcraft.yaml]: Nitpick: this part of the description looks a bit confusing (and yes, I know it comes from our control file, which might need re-visiting too)

File: parts/plugins/x_builddeb.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import os
+
+import snapcraft
+
+# cowboy baby, see https://bugs.launchpad.net/snapcraft/+bug/1772584
+def patch_snapcraft():
+    import snapcraft.internal.common
+    # very hacky but gets the job done for now, right now
+    # SNAPCRAFT_FILES is only used to know what to exclude
+    snapcraft.internal.common.SNAPCRAFT_FILES.remove("snap")
```
Comment (stolowski) [inline parts/plugins/x_builddeb.py]: /o\

**Comment by mvo5** [comment]
Merging this now (because tests are green) and I will do a followup with the fixes.

**Pull request merged**